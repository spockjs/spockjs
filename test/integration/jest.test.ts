import { sync as run } from 'cross-spawn';
import * as path from 'path';

// paths
const resolvePath = path.resolve.bind(path, __dirname);

const root = resolvePath('..', '..');
const tsconfig = resolvePath(root, 'tsconfig.json');

const nodeModules = resolvePath(root, 'node_modules');
const tsNode = resolvePath(nodeModules, '.bin', 'ts-node');
const jestCli = resolvePath(nodeModules, 'jest-cli', 'bin', 'jest');

const cwd = resolvePath('jest');

// specific to windows or timing
const removeVolatileElements = (output: string) =>
  output
    .replace(/ \..+jest\.js/, ' PATH_HIDDEN/jest.js')
    .replace(/. successful test/g, '  successful test')
    .replace(/. failing test/g, '  failing test')
    .replace(/ \(\d*\.?\d+m?s\)/g, '')
    .replace(/Time:\s+\d.*\n/, 'TIME HIDDEN\n');

// tests
test('produces correct output', () => {
  const result = run(
    tsNode,
    [
      '--project',
      tsconfig,
      jestCli,
      '--no-cache',
      '--noStackTrace',
      '--runInBand',
    ],
    { cwd },
  );

  expect(result.status).toBe(1);
  expect(removeVolatileElements(result.output.toString())).toMatchSnapshot();
});
