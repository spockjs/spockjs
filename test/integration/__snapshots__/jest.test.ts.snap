// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`produces correct output 1`] = `
",,FAIL PATH_HIDDEN/jest.js
    successful test
    failing test

    failing test

    assert.equal(received, expected) or assert(received) 
    
    Expected value to be (operator: ==):
      true
    Received:
      false
      
      _assert(x === double(2))
              | |   |         
              | |   4         
              3 false         
      
      [number] double(2)
      => 4
      [number] x
      => 3

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   0 total
TIME HIDDEN
Ran all test suites.
"
`;
